cmd/update.go

```diff
@@ -0,0 +1,483 @@
+		for {
+			hdr, err := tr.Next()
+			if err == io.EOF {
+				break
+			}
+			if err != nil {
+				return err
+			}
+			base := filepath.Base(hdr.Name)
+			if hdr.Typeflag == tar.TypeReg && (base == binaryName || base == binaryNameWin) {
+				return replaceExecutable(binPath, tr, hdr.FileInfo().Mode())
+			}
+		}
+		return errors.New("binary not found in tar.gz")
+
+	default:
+		f, err := os.Open(assetPath)
+		if err != nil {
+			return err
+		}
+		defer f.Close()
+		return replaceExecutable(binPath, f, 0)
+	}
+}
+
+func verifySHA256(filePath, shaPath string) error {
+	info, err := os.Stat(shaPath)
+	if err != nil {
+		return err
+	}
+	if info.Size() > maxChecksumSize {
+		return fmt.Errorf("checksum file too large: %d bytes", info.Size())
+	}
+
+	f, err := os.Open(shaPath)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	scanner := bufio.NewScanner(f)
+	if !scanner.Scan() {
+		return errors.New("empty sha256 file")
+	}
+	hashHex := strings.Fields(scanner.Text())[0]
+	want, err := hex.DecodeString(hashHex)
+	if err != nil || len(want) != sha256.Size {
+		return errors.New("invalid sha256 checksum")
+	}
+
+	bin, err := os.Open(filePath)
+	if err != nil {
+		return err
+	}
+	defer bin.Close()
+
+	h := sha256.New()
+	if _, err = io.Copy(h, bin); err != nil {
+		return err
+	}
+	got := h.Sum(nil)
+
+	if !bytes.Equal(got, want) {
+		return fmt.Errorf("checksum mismatch: expected %s got %s",
+			hashHex, hex.EncodeToString(got))
+	}
+	return nil
+}
+
+var updateCmd = &cobra.Command{
+	Use:   "update",
+	Short: "Download and install the newest Oasis CLI release",
+	Long: `Checks GitHub releases for a newer Oasis CLI version that matches
+your platform, prints the changelog, optionally asks for confirmation,
+then downloads and replaces the current binary.`,
+	Args: cobra.NoArgs,
+	Run: func(_ *cobra.Command, _ []string) {
+		exePath, err := osExecutable()
+		cobra.CheckErr(err)
+		finishPendingWindowsUpdate(exePath)
+
+		apiCtx, cancel := context.WithTimeout(context.Background(), apiTimeout)
+		defer cancel()
+
+		fmt.Println("Checking for updates…")
+		rel, err := fetchLatest(apiCtx)
+		cobra.CheckErr(err)
+
+		current := cliVersion.Software
+		latest := rel.TagName
+
+		isNewer, err := semverDiffers(latest, current)
+		cobra.CheckErr(err)
+		if !isNewer {
+			fmt.Printf("Oasis CLI is already up to date (%s).\n", current)
+			return
+		}
+
+		assetName, assetURL, err := findAsset(rel)
+		cobra.CheckErr(err)
+
+		fmt.Printf("New version %s available (%s).\n", latest, assetName)
+
+		fmt.Printf("\n─ Changelog for %s ─\n%s\n", latest, rel.Body)
+
+		// Ask for confirmation unless -y/--yes used.
+		progress.Confirm("Upgrade now?", "upgrade cancelled")
+
+		dlCtx, cancelDL := context.WithTimeout(context.Background(), downloadTimeout)
+		defer cancelDL()
+
+		assetPath, err := downloadFunc(dlCtx, assetURL)
+		cobra.CheckErr(err)
+		defer func() { _ = os.Remove(assetPath) }()
+
+		if shaURL, ok := findSha256For(rel, assetName); ok {
```

Comment:
Abort if sha256sum not found.
```suggestion
		shaURL, ok := findSha256For(rel, assetName)
		if !ok {
			cobra.CheckErr(...)
		}
```